### PR TITLE
feat(Table): add tooltip for truncated Td, docs(Filter): add filter demos

### LINKED ISSUE #7965: Filter demos: attribute search, filter groups, faceted
Follow up for #6710

Tracking the remaining filter demos: attribute search, filter groups, and faceted filter.

Improvements to current demos (and include in the remaining demos):
- bulk select should account for the filter state of the table data
- selection `th` column is disappearing when there is an empty state being rendered
- menu ids should be updated to be independent
- update tds to use truncate modifier to prevent column resizing on filtering

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #7965

Adds remaining filter demos (attribute search, same select filter group, mixed select filter group, faceted filter).

Updates all filter demos to have the following:
- Bulk select respects filtered state
- Menu ids are unique between demos
- Checkbox column space is maintained when there is an empty state
- `Td`s are truncated to maintain column widths while filtering

While adding the last bullet I also noticed that no tooltip is added when Tds are truncated like Ths should I added that functionality as well.

Other updates in this PR:
- `Td`: Added `tooltip` prop, and when using the truncate modifier, `Td` will attempt to make a default tooltip like `Th`
- `Th`: Updated internal `canDefault` variable name to `canMakeDefaultTooltip` (matched in `Td` as well)
- `MenuToggle`: Added missing ref spread to split button variant of the `MenuToggle`
- `Menu`: added null check in `createNavigableItems` callback that can throw a console error if the menu closes due to the extraKeys handler

### CODE REVIEW

File: packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx
Comment (thatblindgeye) [inline packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx:157]: These need to be updated to `useRef` in all demo files. Currently opening the bulk dropdown doesn't place focus on that first item, and pressing Escape/Tab doesn't close the menu.

File: packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx
Comment (thatblindgeye) [inline packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx:177]: Similarly after updating the `useRef`s mentioned above (I would imagine in all demos), closing the menu with Escape doesn't make focus visually apparent. It looks like `bulkSelectToggleRef` is either undefined or null here. Additionally, we would want to have focus go to the button within the toggle rather than the whole toggle itself.

File: packages/react-table/src/components/TableComposable/Td.tsx
Comment (thatblindgeye) [inline packages/react-table/src/components/TableComposable/Td.tsx:254]: Is this variable name accurate? I'm not sure if it's descriptive for its purpose or if I'm not interpreting it correctly.

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Had a few comments below (one nit, one clarifying question, and a couple issues I noticed).

File: packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx
Comment (kmcfaul) [reply packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx:177]: I'll take another look at the bulk select. Some of the base code was from an older demo so I might have forgotten to update something in changing to the Menu components.

File: packages/react-table/src/components/TableComposable/Td.tsx
Comment (kmcfaul) [reply packages/react-table/src/components/TableComposable/Td.tsx:254]: I matched what was present in `Th`, I'll look at it some more to see if there's a better name.

File: packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx
Comment (kmcfaul) [reply packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx:177]: Turns out the ref wasn't being spread to the split button toggle so I'll update that and put the ref on the overall div (that way a user can choose to return focus to either the checkbox label or toggle button). Unless it make's more sense to put the ref on the toggle button part directly?

**Review by mcarrano — COMMENTED** [review]
@kmcfaul these look great at a glance. But I'll defer to @mmenestr for a thorough design review.

**Comment by mmenestr** [comment]
@kmcfaul looks great overall!! Only noticed one little bug in the last demo, where there should be a badge showing the number of selections - and it seems to work for the first attribute, but not for the second. As soon as you select anything from the second attribute, the badge dissapears.

https://user-images.githubusercontent.com/51165119/192321668-2a140037-1cd3-4492-a938-9b3e5e531539.mov

File: packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx
Comment (thatblindgeye) [reply packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx]: While I think it might make more sense to have focus go solely to the toggle button (as it does with the latest updates), I could also see there being a use-case of placing the ref where you ended up placing it on that overall div.

**Review by thatblindgeye — APPROVED** [review]

**Review by wise-king-sullyman — COMMENTED** [review]
Looks good other than the badge bug in the faceted filter demo that was already mentioned, ready to approve once that's resolved.

**Comment by kmcfaul** [comment]
Fixed the faceted filter badge. @mmenestr @wise-king-sullyman 

Not sure why but the check being inline wasn't quite executing right (it was only looking at the first condition), probably a syntax thing. Moving the check above to a const lets it work properly.

**Review by wise-king-sullyman — APPROVED** [review]

**Review by mmenestr — APPROVED** [review]
Looks great!

File: packages/react-table/src/components/TableComposable/Td.tsx
Comment (mcoker) [inline packages/react-table/src/components/TableComposable/Td.tsx:56]: This block could use some punctuation

Screen Shot 2022-10-03 at 11 09 31 AM

File: packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx
Comment (mcoker) [inline packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx:616]: Do these need pagination in the toolbar if it doesn't do anything?

**Review by mcoker — CHANGES_REQUESTED** [review]
I tried adding `tooltip="tooltip text"` and `tooltip={<div>tooltip text</div>}` to cells and it didn't trigger a tooltip - should that work or is this something else? 

Sortable cells seem to be triggering 2 tooltips.
Screen Shot 2022-10-03 at 11 17 45 AM

And if a cell is sortable and has a help icon, when you mouse off the sort button and to the help button (but keep your mouse in the table cell), one of the cell tooltips stays open, then the help button tooltip shows.
Screen Shot 2022-10-03 at 11 19 08 AM

File: packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx
Comment (kmcfaul) [reply packages/react-core/src/demos/Filters/examples/FilterAttributeSearch.tsx:616]: Design confirmed wanting the paginations in on the previous filter demos PR

File: packages/react-table/src/components/TableComposable/Td.tsx
Comment (kmcfaul) [reply packages/react-table/src/components/TableComposable/Td.tsx:56]: I'll update that (and the existing `Th` one I copied it from)

**Comment by kmcfaul** [comment]
The tooltip prop is for specifying a custom tooltip when a tooltip is rendered (when there is the truncate modifier and the text length overflows). I don't think it forces the tooltip. It should behave the same as the `Th` prop. @nicolethoen @mcoker

**Review by mcoker — APPROVED** [review]
LGTM. Confirmed the double tooltip is a pre-existing issue. Here's an issue for that - https://github.com/patternfly/patternfly/issues/5136

**Pull request merged**
